File: src/game/characters/Trunks.ts
import { BaseFighter } from "../BaseFighter";

export class Trunks extends BaseFighter {
  readonly id = "trunks" as const;
  readonly name = "Trunks";
  readonly greeting = "I will protect this future.";
  readonly signatureAttack = "Burning Attack";
  readonly attackStyle = "orb" as const;
  readonly power = 10500;
  readonly speed = 90;
  readonly defense = 77;
  readonly criticalChance = 0.22;
  readonly baseDamage = 21;
  readonly nativeFacing = "right" as const;
  readonly scale = 0.98;
  readonly palette = {
    skin: "#efae7a",
    hair: "#b58cff",
    primary: "#355fc4",
    secondary: "#202942",
    accent: "#f0d55b",
    aura: "#9fc8ff"
  };
}

File: csharp-reference/Characters/Trunks.cs
using DragonBallArena.Characters.Models;

namespace DragonBallArena.Characters.Characters;

public sealed class Trunks : BaseFighter
{
    public override string Id => "trunks";
    public override string Name => "Trunks";
    public override string SignatureAttack => "Burning Attack";
    public override FighterStats Stats => new(10500, 90, 77, 0.22, 21);
    public override FighterPalette Palette => new(
        Skin: "#efae7a",
        Hair: "#b58cff",
        Primary: "#355fc4",
        Secondary: "#202942",
        Accent: "#f0d55b",
        Aura: "#9fc8ff");
}
